### ISSUE TITLE
"Could not talk to ldap" using openldap access control

### ISSUE THREAD

**Issue by ookongo** [rancher/rancher#5943]
_Describe your issue here_

Hi ,
---

| Useful | Info |
| :-- | :-- |
| Versions | Rancher `v1.1.3` Cattle: `v0.165.8.1` UI: `v1.1.12` |
| Access | `Disabled` |
| Orchestration | `Cattle` |
| Route | `admin-tab.auth.openldap` |

testing the authentication keeps returning:

"Could not talk to ldap"

This is an ipa ldap server.
I can ping the ldap server from the rancher/server container.

Any help would be greatly appreciated.

Thank you,
Othieno.

**Comment by tfiduccia** [comment]
Normally that would mean that one of the fields in Section 1 is not filled out properly.  
1.  Also are you trying with TLS enabled or disabled? If enabled, uncheck and see if you get same error.
2.  Can you give me example of what your fields looks like? (leave out hostname)

**Comment by praseodym** [comment]
I'm seeing the same error when trying to connect to our OpenLDAP server. We use LDAPS on port 636.

The following lines are logged to `cattle-debug.log`, but they're not too useful:
```
2016-10-09 11:56:47,563 INFO  [:] [] [] [] [tp1198425174-14] [io.cattle.platform.pool.PoolConfig  ] Setting Pool [ldap.context.pool] property [jmxNameBase=org.apache.commons.pool2:type=GenericObjectPool,name=] 
2016-10-09 11:56:47,563 INFO  [:] [] [] [] [tp1198425174-14] [io.cattle.platform.pool.PoolConfig  ] Setting Pool [ldap.context.pool] property [jmxNamePrefix=ldap.context.pool] 
2016-10-09 11:56:47,563 INFO  [:] [] [] [] [tp1198425174-14] [io.cattle.platform.pool.PoolConfig  ] Setting Pool [ldap.context.pool] property [maxIdle=50] 
2016-10-09 11:56:47,563 INFO  [:] [] [] [] [tp1198425174-14] [io.cattle.platform.pool.PoolConfig  ] Setting Pool [ldap.context.pool] property [maxTotal=1000] 
2016-10-09 11:56:47,563 INFO  [:] [] [] [] [tp1198425174-14] [io.cattle.platform.pool.PoolConfig  ] Setting Pool [ldap.context.pool] property [maxWaitMillis=1000] 
2016-10-09 11:56:47,563 INFO  [:] [] [] [] [tp1198425174-14] [io.cattle.platform.pool.PoolConfig  ] Setting Pool [ldap.context.pool] property [minEvictableIdleTimeMillis=300000] 
2016-10-09 11:56:47,563 INFO  [:] [] [] [] [tp1198425174-14] [io.cattle.platform.pool.PoolConfig  ] Setting Pool [ldap.context.pool] property [testOnBorrow=true] 
2016-10-09 11:56:47,563 INFO  [:] [] [] [] [tp1198425174-14] [io.cattle.platform.pool.PoolConfig  ] Setting Pool [ldap.context.pool] property [testWhileIdle=true] 
2016-10-09 11:56:47,563 INFO  [:] [] [] [] [tp1198425174-14] [io.cattle.platform.pool.PoolConfig  ] Setting Pool [ldap.context.pool] property [timeBetweenEvictionRunsMillis=60000] 
2016-10-09 11:56:47,567 INFO  [:] [] [] [] [tp1198425174-14] [.i.l.a.LdapServiceContextPoolFactory] Failed to create a service context. 
2016-10-09 11:56:47,567 INFO  [:] [] [] [] [tp1198425174-14] [.i.l.a.LdapServiceContextPoolFactory] ldapserver:636; socket closed 
```

**Comment by aemneina** [comment]
Is your LDAP using a self signed certificate for TLS?

**Comment by praseodym** [comment]
No, we're using a certificate signed by our internal CA. The internal CA was added to the Rancher server as documented (at `/ca.crt`).

**Comment by baptistedonaux** [comment]
Same fail, do you have a solution ?

**Comment by praseodym** [comment]
Unfortunately I do not yet have a solution.

**Comment by aemneina** [comment]
if you guys can provide information Tani was asking for earlier, please reopen.

**Comment by longwuyuan** [comment]
Reopen Please.

Rancher 1.3
TLS Disabled = "Could not talk to ldap"
TLS Enabled = "Could not talk to ldap"


Fields ;
Hostname =  myhost.mydomain.com = "Could not talk to ldap"
IP-Address =  X.X.X.X = "Could not talk to ldap"
Port = 636 and 389 = "Could not talk to ldap"
Service Account Username = uid=testuser1,cn=users,cn=accounts,dc=mycompany,dc=com = "Could not talk to ldap"
Password = <somepassword> = "Could not talk to ldap"

The problem is the ca.crt for the java process doing ldap ;

- I check logs if ca.crt was added to container. It was.

# docker logs rancher-server > logs
...
..
# grep ca.crt logs
Adding ca.crt to Certs.
# 

I did a tcpdump on the ipa server to see connections. I saw connections from rancher. But still "Could not talk to ldap".

I 'exec'd into the container, installed ldap-utils and did a ldapsearch on my ipa server. It worked. Success. I edited /etc/ldap/ldap.conf to point to /ca.crt as that is what ldap-utils looks for .


So it is the java process fro LDAP SSL not seeing the ca.crt. If I don't use TLS, then success.

SO please tell where the java process for LDAP is looking for the ca.crt.  I think if I mount the ca.crt in the right, place, this will work.

Hope you respond asap as this is basic and simple to tell where java wants the crt file.

**Comment by longwuyuan** [comment]
This is on rancher:stable as of today

**Comment by deniseschannon** [comment]
@longwuyuan `rancher/server:stable` is the same as v1.2.2, and it is fixed in `rancher/server:latest`, v1.3.0.

**Comment by aemneina** [comment]
@longwuyuan as @deniseschannon mentioned this should be fixed in 1.3.0, see #6891 for workaround steps for 1.2.x.
